### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by seperalta22**
# Blog App - processing data in models

## Summary

In this PR, I created four models User, Comment, Like, and Post, each of them has some methods and where linked using foreign ids.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :comments, foreign_key: :author_id
+  has_many :posts, foreign_key: :author_id
+  has_many :likes, foreign_key: :user_id
+
+  def recent_posts
+    posts.limit(5).order(created_at: :desc)
+  end
```
Comment (nahnahdev) [inline app/models/user.rb]: - [x] Great job so far @SergioPeralta22 There is just one change I will suggest you make and you'll be on your way to the next milestone :wink: 
- [x] Great job creating the `recent_posts` method but per the project requirements, this method needs to return the 3 most recent posts. Currently, yours returns 5 so kindly fix that :+1:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @SergioPeralta22 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Has `User`, `Post`, `Comment`, and `Like` models inside `app/models/`  directory :heavy_check_mark: 
- The associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). The associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). :heavy_check_mark: 
- Post model has a method that updates the posts counter for a user :heavy_check_mark: 
- Post model has a method that returns the 5 most recent comments for a given post :heavy_check_mark: 
- Comment model has a method that updates the comments counter for a post :heavy_check_mark: 
- Like model has a method that updates the likes counter for a post :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (rloterh) [reply app/models/user.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Sergio,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvements to the data models!
🎲 Professional ReadMe!

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**